packages/framework/framework/src/container.ts

```diff
@@ -7,11 +7,17 @@ import { AwilixContainer, ResolveOptions } from "awilix"
  */
 export interface ModuleImplementations {}
 
+type ToAsyncMethods<T extends object> = {
+  [K in keyof T]: T[K] extends (...args: infer A) => infer R
+    ? T[K] extends (...args: A) => Promise<infer PR> ? T[K] : (...args: A) => Promise<R>
+    : T[K]
+}
+
```

Comment:
```suggestion
type TransformObjectMethodToAsync<T extends object> = {
  [K in keyof T]: T[K] extends (...args: infer A) => infer R
    ? (...args: A) => Promise<Awaited<R>>
    : T[K] extends object ? TransformObjectMethodToAsync<T[K]> : T[K];
}
```

Very nice, I would just simplify it a bit and also take into consideration nested objects, here you want to change the method of the interfaces of ModuleImplementations for which the props are interfaces :)

Reply:
In regards to its location, I would eventually move it to core/types/src/common and maybe name it differently like the proposal above? cc @olivermrbl

Reply:
@adrien2p, think that makes sense. Should we get this merged?

Reply:
It is good to me 👌